test/parallel/test-runner-exit-code.js

```diff
@@ -37,14 +46,20 @@ if (process.argv[2] === 'child') {
   assert.strictEqual(child.status, 1);
   assert.strictEqual(child.signal, null);
 
-  child = spawnSync(process.execPath, [__filename, 'child', 'never_ends']);
-  assert.strictEqual(child.status, 1);
-  assert.strictEqual(child.signal, null);
-  if (common.isWindows) {
-    common.printSkipMessage('signals are not supported in windows');
-  } else {
-    const stdout = child.stdout.toString();
-    assert.match(stdout, /not ok 1 - never ending test/);
-    assert.match(stdout, /# cancelled 1/);
-  }
+  runAndKill(fixtures.path('test-runner', 'never_ending_sync.js')).then(common.mustCall(async (stdout) => {
+    if (common.isWindows) {
+      common.printSkipMessage('signals are not supported in windows');
+    } else {
+      assert.match(stdout, /not ok 1/);
+      assert.match(stdout, /# cancelled 1\n/);
+    }
+  }));
+  runAndKill(fixtures.path('test-runner', 'never_ending_async.js')).then(common.mustCall(async (stdout) => {
+    if (common.isWindows) {
+      common.printSkipMessage('signals are not supported in windows');
+    } else {
+      assert.match(stdout, /not ok 1/);
+      assert.match(stdout, /# cancelled 1\n/);
+    }
+  }));
```

Comment:
Why not make the `assert.match` calls part of `runAndKill` to avoid the repetition? Otherwise, you need to add another `.then(common.mustCall())` to ensure no errors is swalloded.